src/components/CommonForm/index.js

```diff
@@ -44,14 +40,44 @@ const CommonForm = ({ form, title, submit_title, submit_body }) => {
       behavior: "smooth"
     });
   };
-
+  // Validating role
+  const isValidRole = (value) => {
+    if (!value) {
+      setValidateRole(true);
+      return false;
+    } else {
+      setValidateRole(false);
+      return true;
+    }
+  };
+  /*
+  Validating Email ----
+  1. Check for email if it satisfies the standard forn for email
+  2. Check for anonymous email address/domains. Currently It is checking for only 4 domains, we can add more to regex expression as we move ahead.
+  */
+  const isValidEmail = (value) => {
+    const validEmail = /^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}$/.test(value);
+    if (!validEmail) {
+      setValidateEmail(true);
+      setErrorEmail("Please enter a valid email address");
+      return false;
+    } else if (validEmail && /.*(?=.*@(?:duck\.com|protonmail\.com|anonaddy\.me|tuta\.io)).*/.test(value)) {
+      setValidateEmail(true);
+      setErrorEmail("Use of this email address is not allowed. Please enter a non-anonymous email address / domain.");
+      return false;
+    } else {
+      setValidateEmail(false);
+      setErrorEmail("");
+      return true;
+    }
+  };
   useEffect(() => {
     if (submit) {
       axios.post("https://hook.us1.make.com/s4jo47a8ydq6uk7gg16wuukgr4l52p8c", {
         memberFormOne,
       });
     }
-  }, [submit]);
+  }, [submit, errorEmail, validateEmail, validateRole]);
```

Comment:
I couldn't get why these additional states needed to have an effect. Like whenever I submit a form only then the inner if will run and the request will be made..

Reply:
I added this since useEffect does re-rendering of the component so if the validation fails then submit's value is not gonna change and the component won't get re-rendered.

Reply:
You're not getting my point

Let me ask you some simple questions..
1. When do you want this post request to be made?
2. When is the value of the `errorEmail` being set?

Reply:
> I added this since useEffect does re-rendering of the component so if the validation fails then submit's value is not gonna change and the component won't get re-rendered.

Also, the error message is in a state so whenever it changes there is going to be a re-render.. it doesn't need to be given an effect.